[Bug] 1.1.3 Failed to generate image after upgrading 1.2.0

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

1.2.0-rc04

### What's Wrong?

When I upgrade the cluster from 1.1.3 to 1.2.0, the fe log shows that image generation failed: 

`2022-12-09 21:57:29,936 WARN (leaderCheckpointer|144) [MetaWriter.lambda$write$2():119] failed to write meta module: db
java.lang.reflect.InvocationTargetException: null
    at sun.reflect.GeneratedMethodAccessor68.invoke(Unknown Source) ~[?:?]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
    at org.apache.doris.persist.meta.MetaWriter.lambda$write$2(MetaWriter.java:117) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.lambda$setDelegate$0(MetaWriter.java:85) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.doWork(MetaWriter.java:93) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.write(MetaWriter.java:115) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:2010) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:1990) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.doCheckpoint(Checkpoint.java:131) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.runAfterCatalogReady(Checkpoint.java:76) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.doCheckpoint(Checkpoint.java:131) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.runAfterCatalogReady(Checkpoint.java:76) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.common.util.Daemon.run(Daemon.java:116) ~[doris-fe.jar:1.0-SNAPSHOT]
Caused by: java.lang.NullPointerException
2022-12-09 21:57:29,937 ERROR (leaderCheckpointer|144) [Checkpoint.doCheckpoint():151] Exception when generate new image file
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
    at org.apache.doris.persist.meta.MetaWriter.lambda$write$2(MetaWriter.java:120) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.lambda$setDelegate$0(MetaWriter.java:85) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.doWork(MetaWriter.java:93) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.write(MetaWriter.java:115) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:2010) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:1990) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.doCheckpoint(Checkpoint.java:131) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.runAfterCatalogReady(Checkpoint.java:76) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.common.util.Daemon.run(Daemon.java:116) ~[doris-fe.jar:1.0-SNAPSHOT]
Caused by: java.lang.reflect.InvocationTargetException
    at sun.reflect.GeneratedMethodAccessor68.invoke(Unknown Source) ~[?:?]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
    at org.apache.doris.persist.meta.MetaWriter.lambda$write$2(MetaWriter.java:117) ~[doris-fe.jar:1.0-SNAPSHOT]
    ... 9 more
Caused by: java.lang.NullPointerException
2022-12-09 21:57:29,941 WARN (leaderCheckpointer|144) [Checkpoint.runAfterCatalogReady():78] failed to do checkpoint.
org.apache.doris.common.CheckpointException: java.lang.reflect.InvocationTargetException
    at org.apache.doris.master.Checkpoint.doCheckpoint(Checkpoint.java:155) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.runAfterCatalogReady(Checkpoint.java:76) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.common.util.Daemon.run(Daemon.java:116) ~[doris-fe.jar:1.0-SNAPSHOT]
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
    at org.apache.doris.persist.meta.MetaWriter.lambda$write$2(MetaWriter.java:120) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.lambda$setDelegate$0(MetaWriter.java:85) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.doWork(MetaWriter.java:93) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.write(MetaWriter.java:115) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:2010) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:1990) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.doCheckpoint(Checkpoint.java:131) ~[doris-fe.jar:1.0-SNAPSHOT]
    ... 3 more
    Caused by: java.lang.reflect.InvocationTargetException
    at sun.reflect.GeneratedMethodAccessor68.invoke(Unknown Source) ~[?:?]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
    at org.apache.doris.persist.meta.MetaWriter.lambda$write$2(MetaWriter.java:117) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.lambda$setDelegate$0(MetaWriter.java:85) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.doWork(MetaWriter.java:93) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.persist.meta.MetaWriter.write(MetaWriter.java:115) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:2010) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.catalog.Env.saveImage(Env.java:1990) ~[doris-fe.jar:1.0-SNAPSHOT]
    at org.apache.doris.master.Checkpoint.doCheckpoint(Checkpoint.java:131) ~[doris-fe.jar:1.0-SNAPSHOT]
    ... 3 more
Caused by: java.lang.NullPointerException`

When I want to dump metadata from memory, I also report an error: 
`{"msg":"Internal Error","code":500,"data":"java.lang.reflect.InvocationTargetException","count":0}`


### What You Expected?

Can write to the image normally.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/fe/fe-core/src/main/java/org/apache/doris/catalog/DataProperty.java
+++ b/fe/fe-core/src/main/java/org/apache/doris/catalog/DataProperty.java
@@ -22,9 +22,11 @@ import org.apache.doris.common.FeMetaVersion;
 import org.apache.doris.common.io.Text;
 import org.apache.doris.common.io.Writable;
 import org.apache.doris.common.util.TimeUtils;
+import org.apache.doris.persist.gson.GsonPostProcessable;
 import org.apache.doris.persist.gson.GsonUtils;
 import org.apache.doris.thrift.TStorageMedium;
 
+import com.google.common.base.Strings;
 import com.google.gson.annotations.SerializedName;
 
 import java.io.DataInput;
@@ -32,7 +34,7 @@ import java.io.DataOutput;
 import java.io.IOException;
 import java.util.Objects;
 
-public class DataProperty implements Writable {
+public class DataProperty implements Writable, GsonPostProcessable {
     public static final DataProperty DEFAULT_DATA_PROPERTY = new DataProperty(
             "SSD".equalsIgnoreCase(Config.default_storage_medium) ? TStorageMedium.SSD : TStorageMedium.HDD
     );
@@ -126,7 +128,7 @@ public class DataProperty implements Writable {
 
         return this.storageMedium == other.storageMedium
                 && this.cooldownTimeMs == other.cooldownTimeMs
-                && this.storagePolicy.equals(other.storagePolicy);
+                && Strings.nullToEmpty(this.storagePolicy).equals(Strings.nullToEmpty(other.storagePolicy));
     }
 
     @Override
@@ -136,5 +138,11 @@ public class DataProperty implements Writable {
         sb.append("cool down[").append(TimeUtils.longToTimeString(cooldownTimeMs)).append("]. ");
         sb.append("remote storage policy[").append(this.storagePolicy).append("]. ");
         return sb.toString();
+    }
+
+    @Override
+    public void gsonPostProcess() throws IOException {
+        // storagePolicy is a newly added field, it may be null when replaying from old version.
+        this.storagePolicy = Strings.nullToEmpty(this.storagePolicy);
     }
 }

